Makefile

```diff
@@ -140,20 +140,9 @@ bddtests: python
 	@tox -qe bddtests
 
 # Tell make how to compile requirements/*.txt files.
-#
-# `touch` is used to pre-create an empty requirements/%.txt file if none
-# exists, otherwise tox crashes.
-#
-# $(subst) is used because in the special case of making requirements.txt we
-# actually need to touch dev.txt not requirements.txt and we need to run
-# `tox -e dev ...` not `tox -e requirements ...`
-#
-# $(basename $(notdir $@))) gets just the environment name from the
-# requirements/%.txt filename, for example requirements/foo.txt -> foo.
 requirements/%.txt: requirements/%.in
-	@touch -a $(subst requirements.txt,dev.txt,$@)
-	@tox -qe $(subst requirements,dev,$(basename $(notdir $@))) --run-command 'pip --quiet --disable-pip-version-check install pip-tools'
-	@tox -qe $(subst requirements,dev,$(basename $(notdir $@))) --run-command 'pip-compile --allow-unsafe --quiet $(args) $<'
+	pyenv exec pip --quiet --disable-pip-version-check install --upgrade pip-tools
+	pyenv exec pip-compile --allow-unsafe --quiet $(args) $<
```

Comment:
### Polluting the user's Python installations

This will install `pip-tools` directly into each pyenv Python instance. It'll get installed at `$(pyenv root)/versions/X.Y.Z/lib/python3.8/site-packages/piptools/` for each version of Python used by one of our apps (`$(pyenv root)` is `~/.pyenv` by default). This is kind of polluting the user's pyenv installations: there's only one `$(pyenv root)/versions/X.Y.Z/` on the user's system for each version of Python, and the user may use pyenv for things other than Hypothesis projects.

We already [install tox in the same way](https://github.com/hypothesis/lms/blob/536d3c623e7eefd73a144bb9aa0ed07e3e4fb9de/bin/install-python#L43).

Ideally I think we'd use `pyenv exec python -m venv` to create a venv somewhere (probably in a `$(pyenv root)/versions/` subdir) and install `pip-tools` into that venv. And do the same to create another venv to install `tox` into. But that'd require some scripting as we'd need to check whether the venvs already exists and create them if they don't.

Creating venvs in `$(pyenv root)/versions/` is what [pyenv-virtualenv](https://github.com/pyenv/pyenv-virtualenv) (an official pyenv plugin) does (but I don't think we want to use pyenv-virtualenv as then we'd have to get everyone to install it).

Alternatively we could `pyenv exec pip install pipx` directly into the pyenv Python installation and then `pyenv exec pipx install tox pip-tools`. This way we're polluting the pyenv Python installation with `pipx` only and we can then use `pipx` to install tools into isolated venvs for us instead of having to script that ourselves.

### Not pinning `pip-tools`

The code as written always installs or upgrades to the latest version of `pip-tools` before running `pip-compile`. Our version of `pip-tools` isn't pinned so new releases of `pip-tools` could break `make requirements`. We could try to address this by having a `requirements/pip-tools.txt` file and installing that in a manually-created venv (not a tox one). I suspect this may not be worth the bother and I don't think it's compatible with the `pipx` suggestion above